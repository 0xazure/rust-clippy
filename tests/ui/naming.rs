// Copyright 2018 The Rust Project Developers. See the COPYRIGHT
// file at the top-level directory of this distribution.
//
// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
// option. This file may not be copied, modified, or distributed
// except according to those terms.

pub struct MyStruct {
    id: usize
}

impl MyStruct {
    pub fn get_id(&self) -> usize {
        self.id
    }

    pub fn get(&self) -> usize {
        self.id
    }

    pub fn get_mut(&mut self) -> usize {
        self.id
    }

    pub fn get_unchecked(&self) -> usize {
        self.id
    }

    pub fn get_unchecked_mut(&mut self) -> usize {
        self.id
    }

    pub fn get_ref(&self) -> usize {
        self.id
    }
}

fn main() {
   let mut s = MyStruct { id: 42 };
   s.get_id();
   s.get();
   s.get_mut();
   s.get_unchecked();
   s.get_unchecked_mut();
   s.get_ref();
}
