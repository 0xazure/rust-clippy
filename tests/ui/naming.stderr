error: prefixing a getter with `get_` does not follow naming conventions
  --> $DIR/naming.rs:15:5
   |
LL | /     pub fn get_id(&self) -> usize {
LL | |         self.id
LL | |     }
   | |_____^
   |
   = note: `-D clippy::getter-prefix` implied by `-D warnings`

error: aborting due to previous error

